File: backend/ms-paciente/src/main/java/cl/rednorte/ms_paciente/dto/PatientDTO.java
package cl.rednorte.ms_paciente.dto;

import cl.rednorte.ms_paciente.model.status.Gender;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PatientDTO {

    private String id;

    // RUN u otro identificador
    private String identifierType;
    private String identifierValue;

    private String firstName;
    private String lastName;

    private Gender gender;

    private String phone;
    private String email;
    private String address;
    private Boolean active;

    private CoverageDTO coverage;
}

File: frontend/src/remotes/dtos/patient.dto.ts
// Matches backend cl.rednorte.ms_paciente.dto.CoverageDTO
export interface CoverageDTO {
  id?: string;
  type: 'FONASA' | 'ISAPRE' | string;
  provider?: string;
  code?: string;
  name?: string;
  description?: string;
  [key: string]: any;
}

// Matches backend cl.rednorte.ms_paciente.dto.PatientDTO
export interface PatientDTO {
  names?: any;
  identifiers?: any;
  id?: string;
  identifierType: 'RUN' | 'PASSPORT' | string;
  identifierValue: string;
  firstName?: string;
  lastName?: string;
  birthDate?: string;
  gender?: 'MALE' | 'FEMALE' | 'OTHER' | 'UNKNOWN';
  phone?: string;
  email?: string;
  address?: string;
  active?: boolean;
  coverage?: CoverageDTO;
  prevision?: string;
  [key: string]: any;
}

export type CreatePatientDTO = Omit<PatientDTO, 'id'> & {
  birthDate?: string;
  prevision?: string;
  [key: string]: any;
};


/** Returns a human-readable label for a coverage, e.g. "Fonasa", "Isapre - Consalud" */
export function coverageLabel(dto: CoverageDTO): string {
  const base =
    dto.type === 'FONASA' ? 'Fonasa' :
    dto.type === 'ISAPRE' ? 'Isapre' :
    dto.type;
  return dto.provider ? `${base} - ${dto.provider}` : base;
}
